extensions/amp-script/0.1/amp-script.js

```diff
@@ -157,28 +167,14 @@ export class AmpScript extends AMP.BaseElement {
       );
     }
 
-    return getElementServiceForDoc(this.element, TAG, TAG).then((service) => {
-      this.setService(/** @type {!AmpScriptService} */ (service));
-    });
-  }
-
-  /**
-   * @override
-   */
-  onLayoutMeasure() {
-    if (this.layoutCompleted_ || this.reportedZeroSize_) {
-      return;
-    }
-
-    const {width, height} = this.getLayoutSize();
```

Comment:
We could still send a warning on each mount using the `measureSize` from the `size-observer.js`.

Reply:
I want the component to be laid out regardless of size. Is that possible while deferring build?

Reply:
It's all based on the `IntersectionObserver`. If it gives `isIntersecting == true` - then yes. But to note a couple of things:

1. I'm not convinced this is a very useful warning, tbh.
2. This piece of code was solely for warning (just in case). And it was for a case when `<amp-script>` should be showing in DOM but initial content makes it zero size, which `Resources` ignored. It looks like the `nodom` is explicitly a different case. For the `<amp-script>` that should be shown a similar rules might still be a good idea.
3. `isIntersecting` and `size != 0` are not the same thing for the `IntersectionObserver`. The `display:none` would always be `!isIntersecting`. But zero-size might still be `isIntersecting == true`.

I guess what I'm trying to say - let's try it out and see :)

Reply:
Would you expect a very far out-of-viewport amp-script to run? Personally I think so.

Reply:
I'd not and would want to see a good reason why it should. I think of such `<amp-script>` instances as a "better version of an iframe". I'm ready to concede that a `<amp-script nodom>` should be different, however.

Reply:
Also, the best I see, the current `amp-script` implementation agrees with that assessment. But I might be missing something.